### DISCUSSION TITLE
How to enforce NOT NULL constraint for a String[] field?

### DISCUSSION THREAD

**Discussion by xddz9** [prisma/prisma#3999] [Q&A] [answered]
Take the following model:

```
model Person {
    name: String
    pets: String[]
}
```

When generating the prisma client, Typescript comes up with types of `{ name: string, pets?: string[] }`, instead of `{ name: string, pets: string[] }`.

How can I enforce pets to always be an array, even if an empty one, and disallow NULL values?

Even if I manually set a not null constraint on the pets field in Postgres and run `prisma generate`, my schema doesn't change and the client still allows for null/undefined values. Is this by design or am I doing something wrong here?

(PostgreSQL 12.4)

**Comment by ryands17** [comment, marked as answer]
Currently it's not possible via Migrate and do follow this issue https://github.com/prisma/migrate/issues/519 for updates
As a workaround, you would explicitly need to explicitly set this in the database and add it as a default value

Something like (example table):

```sql
create table "user" (
	id serial primary key,
	name text,
	addresses text[] not null default '{}'::text[]
)
```

This will not cause an error when even when you do not pass the array as it is set to default.

**Reply by xddz9** [reply]
Got it! Thanks for the workaround and for pointing out the relevant issue, I'll keep an eye on that. :)

**Comment by jbdemonte** [comment]
any update?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
